java/src/shared/model/game/Game.java

```diff
@@ -813,6 +813,15 @@ public void discardCards(int playerIndex, List<ResourceType> cards) throws Playe
         if (canDiscardCards(playerIndex) && this.turnTracker.canDiscard()) {
             playerManager.discardResourceType(playerIndex, cards);
         }
+        boolean change = true;
```

Comment:
This will be more readable if you call it `allPlayersDiscarded` or something like that